### PR TITLE
Move Datadog to install with Flux

### PR DESCRIPTION
**Pull request by phillebaba**

### CODE REVIEW

File: modules/kubernetes/datadog/main.tf [hunk 1]
```diff
@@ -14,15 +14,149 @@ terraform {
...
+  namespace: flux-system
+spec:
+  interval: 5m
+  sourceRef:
+    kind: GitRepository
+    name: flux-system
+  path: ./platform/we-dev-aks1/datadog
+  prune: true
+  validation: client
+ EOF
```
Comment (simongottschlag) [inline modules/kubernetes/datadog/main.tf:39]: I don't like doing these as strings. Either HCL to YAML or using templates. This feels like things like number of spaces will cause issues.

File: modules/kubernetes/datadog/templates/platform.yaml.tpl [hunk 2]
```diff
@@ -1,11 +1,55 @@
...
+spec:
+  chart:
+    spec:
+      chart: datadog-operator
+      sourceRef:
+        kind: HelmRepository
+        name: datadog
+      version: 0.8.0
+  values:
+    crds: CreateReplace
```
Comment (phillebaba) [inline modules/kubernetes/datadog/templates/platform.yaml.tpl:33]: Remove line

File: modules/kubernetes/datadog/variables.tf [hunk 3]
```diff
@@ -32,5 +37,5 @@ variable "namespace_include" {
 variable "apm_ignore_resources" {
   description = "The resources that shall be excluded from APM"
   type        = list(string)
-  default     = []
+  default     = [""]
```
Comment (phillebaba) [inline modules/kubernetes/datadog/variables.tf:40]: Why the empty string in list?

**Comment by CalleB3** [comment]
Moved to #972

**Pull request closed without merging**